grpc-for-smart-contracts.md

```diff
@@ -0,0 +1,263 @@
+# gRPC for interacting with Smart Contracts
+
+*Documentation of our gRPC API and data types related to smart contracts.*
+
+Documents the following gRPC calls:
+- `GetAccountInfo`
+- `GetInstances`
+- `GetInstanceInfo`
+- `SendTransaction`
+
+A typical workflow is:
+
+1. Deploy a wasm module by sending a `DeployModule` transaction.
+2. Creating a contract instance of the module by sending a `InitContract`
+   transaction.
+3. Finding the address of the contract instance via `GetInstances` and
+   `GetInstanceInfo`.
+4. Updating the contract by sending a `Update` transaction.
+
+## `GetAccountInfo`
+
+*Retrieve information about an account.*
+
+```protobuf 
+rpc GetAccountInfo(GetAddressInfoRequest) returns (JsonResponse) {}
+
+message GetAddressInfoRequest {
+  string block_hash = 1;
+  string address = 2;
+}
+```
+
+**Input:**
+
+- `block_hash`: A null-terminated base16 encoding of a block hash. 
+- `address`: A null-terminated base58 encoding (same format as returned by `GetAccountList`) of an account address. 
+
+**Output:**
+
+- A JSON object containing information about the given address at the given block.
+  - Example:
+
+    ```json
+    {"accountAmount":"10000000000000","accountNonce":1,"accountEncryptedAmount":{"incomingAmounts":[],"selfAmount":"c00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000c00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000c00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000c00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000","startIndex":0},"accountEncryptionKey":"993fdc40bb8af4cb75caf8a53928d247be6285784b29578a06df312c28854c1bfac2fd0183967338b578772398d4120193c96e1a644ecb148615ec0940cdf30e2f373cc470f393231a4be681b8cbb27edf0b67fda7290a3545d62dc8038d1fa7","accountReleaseSchedule":{"schedule":[],"total":"0"},"accountCredentials":{"0":{"value":{"contents":{"ipIdentity":0,"regId":"93c96e1a644ecb148615ec0940cdf30e2f373cc470f393231a4be681b8cbb27edf0b67fda7290a3545d62dc8038d1fa7","policy":{"revealedAttributes":{},"createdAt":"202103","validTo":"202203"},"credentialPublicKeys":{"keys":{"0":{"verifyKey":"6bb6fb860514c7a1002fc9c3f3a9f7a7a518d54d0437b7dd7e9ba29c65263cd8","schemeId":"Ed25519"},"1":{"verifyKey":"80db411ba83f282ebe7ab3c037e481c72c460de1de87bc57a0b85a7dd347bd76","schemeId":"Ed25519"},"2":{"verifyKey":"c14a3e621d7ff5f1302a0b5c9371fbe8429e8db33b1ff33b962e468a901c159a","schemeId":"Ed25519"}},"threshold":2}},"type":"initial"},"v":0}}}
```

Comment:
Can this be pretty printed?

It's also a bit unclear from this what fields are mandatory, and which  ones are optional.